controllers/base_controller.py

```diff
@@ -48,16 +51,24 @@ def _render_cache_key(cls, cache_key):
             cls.CACHE_VERSION,
             tba_config.CONFIG["static_resource_version"])
 
+    def _add_admin_bar(self, html):
+        if self._is_admin:
+            self.template_values["cache_key"] = self.cache_key
+            self.template_values["user_bundle"] = self._user_bundle
+            admin_bar = jinja2_engine.render('admin_bar.html', self.template_values)
+            return html.replace('<!-- Admin Bar -->', admin_bar)
```

Comment:
Could you modify the base template to include something like

```
{% if should_show_admin_bar %}
{% include "admin_bar.html" %}
{% endif %}
```

And put the value for `should_show_admin_bar` in `self.template_values`?